dialogue-core/src/main/java/com/palantir/dialogue/core/DialogueChannel.java

```diff
@@ -110,6 +107,9 @@ public void updateUris(Collection<String> uris) {
                 ImmutableList.copyOf(limitedChannelByUri.values()),
                 random,
                 channelName));
+
+        // some queued requests might be able to make progress on a new uri now
+        queuedChannel.schedule();
```

Comment:
as mentioned in the other PR, this seems like a strange thing for the DialogueChannel to have to understand (feels very leaky)

Reply:
Ok so theoretically yes dialogue is just one giant stack of Channels all wrapping each other like a big onion and at the top level we only interact with the outer layer, but in practise I think this limits the functionality that can be built.

Given that this is entirely encapsulated inside DialogueChannel, we at least have the freedom to tweak this as nothing is leaked to users

Reply:
In my mind, this is the sort of leakiness that leads to the complexity of CJR. In the original queued implementation, I could describe it's behavior as "It will schedule new requests when they are first submitted, a request completes, and every 100ms". Now I have to describe it as "When they're submitted, a request completes, or when uris are reloaded" which now brings in behavior that has nothing to do with queueing. I think small things like this don't seem bad in isolation, but make understanding the overall behavior extremely difficult because each of the layers has an impact on the others.

Reply:
I would make the same argument against the scheduler :-)

I think small things like the scheduled executor don't seem bad in isolation, but make understanding the overall behavior extremely difficult because each of the layers has an impact on the others.

Happy to chat if you like.

Reply:
Would be more than happy to chat. The thing that scares me is that the next time we add some functionality to update the underlying channels, we'll have to keep in mind that we need to keep poking the queued channel. The arch of this currently means that anytime an underlying channel is updated to increase it's capacity, it should also poke the queued channel which seems like a very strange thing for them to have to think about.